### ISSUE TITLE
Service CIDR pre-check fails if `blackhole` routes exist

### ISSUE THREAD

**Issue by gdemonet** [scality/metalk8s#3349]
**Component**: salt, networking

**What happened**:

Calico creates a "blackhole" route with the BIRD protocol for the CIDR of ClusterIPs assigned to a host, so that it can advertise it to its BGP peers.
Such routes are not properly recognized by Salt's `network.routes` execution method, which we use to check whether a route exists for the Service CIDR.
As such, re-running bootstrap on a node with such a "blackhole" route will fail with a traceback similar to the following:

```
Failure while running step 'Checking local node'

Command: check_local_node

Output:

<< BEGIN >>
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
ValueError: 'blackhole' does not appear to be an IPv4 or IPv6 network
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3002.6', 'console_scripts', 'salt-call')()
  File "/usr/lib/python3.6/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 112, in run
    ret = self.call()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 220, in call
    self.opts, data, func, args, kwargs
  File "/usr/lib/python3.6/site-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/var/cache/salt/minion/extmods/modules/metalk8s_checks.py", line 52, in node
    service_route_ret = route_exists(destination=service_cidr, raises=False)
  File "/var/cache/salt/minion/extmods/modules/metalk8s_checks.py", line 411, in route_exists
    all_routes = __salt__["network.routes"](family="inet")
  File "/usr/lib/python3.6/site-packages/salt/modules/network.py", line 1769, in routes
    routes_ = _ip_route_linux()
  File "/usr/lib/python3.6/site-packages/salt/modules/network.py", line 554, in _ip_route_linux
    address_mask = convert_cidr(comps[0])
  File "/usr/lib/python3.6/site-packages/salt/modules/network.py", line 1237, in convert_cidr
    cidr = calc_net(cidr)
  File "/usr/lib/python3.6/site-packages/salt/modules/network.py", line 1260, in calc_net
    return __utils__["network.calc_net"](ip_addr, netmask)
  File "/usr/lib/python3.6/site-packages/salt/utils/network.py", line 1109, in calc_net
    return str(ipaddress.ip_network(ipaddr, strict=False))
  File "/usr/lib64/python3.6/ipaddress.py", line 84, in ip_network
    address)
ValueError: 'blackhole' does not appear to be an IPv4 or IPv6 network
```

**What was expected**:

Since such routes are expected, our checks should not fail when encountering them.

**Steps to reproduce**:

Re-running bootstrap on a functional deployment should exhibit the same error.
(Note that for some reason, VMs in our internal cloud, used by our CI, do not show such routes, hence do not fail when re-running bootstrap).

**Resolution proposal** (optional):

Implement our own parsing of the kernel routing table, instead of relying on Salt (also open an upstream ticket to fix this behaviour), and ignore such `blackhole` routes.

**Comment by alexandre-allard** [comment]
Opened the issue upstream https://github.com/saltstack/salt/issues/60126

**Comment by alexandre-allard** [comment]
Fixed by #3352
